src/db/collections/block.rs

```diff
@@ -188,6 +189,30 @@ impl BlockCollection {
             .map_ok(|BlockIdResult { block_id }| block_id))
     }
 
+    /// Get the past-cone of a milestone in "White Flag" order.
+    pub async fn get_pastcone_in_white_flag_order(&self, index: MilestoneIndex) -> Result<Vec<BlockId>, Error> {
+        #[derive(Deserialize)]
+        struct BlockIdResult {
+            id: BlockId,
+        }
+
+        let block_ids = self
+            .aggregate::<BlockIdResult>(
+                vec![
+                    doc! { "$match": { "metadata.referenced_by_milestone_index": index } },
```

Comment:
Shouldn't we provide the white flag of the requested block and only get blocks with `"white_flag_index": { "$lt": index }`?

Reply:
That would only be possible if we would have pre-computed and stored all the hashes of the merkle tree. Currently we need to do that on request and I think that's probably the better approach.